### PR TITLE
Update: Sync approved PRs to internal codebase

### PR DESCRIPTION
**Pull request by laukik-hase**
### Trigger workflow on `pull_request_target` event
- Command comment `sha=1a2b3c4` to be followed by affixing of the appropriate label:
        - `PR-Sync-Merge` | `PR-Sync-Rebase` | `PR-Sync-Update`
- New feature: `PR-Sync-Update` -> update the internal branch with new commits on PR
- Labelling and command comment must be done by the same user
- Commit SHA ID specified must match with the most recent commit in the PR fork branch
- Updated README.md with new instructions

### CODE REVIEW

**Review by mahavirj — APPROVED** [review]

**Comment by mahavirj** [comment]
@dobairoland I would like to merge this PR, please check if you have any comments here

File: github_pr_to_internal_pr/README.md [hunk 1]
```diff
@@ -1,22 +1,19 @@
...
-4. The '/merge' command will create an internal PR by fetching the Github PR branch head and pushing it to the internal remote (for relatively new PRs)
+1. When the PR is ready to merge after reviewing, the user needs to add a comment on the PR of the type `sha=1a2b3c4` where `sha` specifies the *SHA1 hash* (long or short) of the **most recent** commit to be merged/updated.
+2. The PR is then affixed with a label `PR-Sync-Merge`, `PR-Sync-Rebase` or `PR-Sync-Update`, triggering the action of syncing the PR. (Labels can only be affixed by a user with an access level > [TRIAGE](https://docs.github.com/en/organizations/managing-access-to-your-organizations-repositories/repository-permission-levels-for-an-organization#permission-levels-for-repositories-owned-by-an-organization) in Github)
+3. The `PR-Sync-Merge` label will create an internal PR by fetching the Github PR branch head and pushing it to the internal remote (for new PRs - close to the current `master` branch). This should be the preferred strategy while syncing.
+4. The `PR-Sync-Merge` label will create an internal PR by fetching the Github PR branch head and pushing it to the internal remote (for relatively new PRs).
```
Comment (kumekay) [inline github_pr_to_internal_pr/README.md:10]: Should it be 'PR-Sync-Rebase'?

**Review by kumekay — COMMENTED** [review]
LGTM, except one small typo

**Review by dobairoland — APPROVED** [review]
LGTM

**Pull request merged**